File: app/src/main/java/com/alexkn/syntact/data/dao/base/BaseDao.java
package com.alexkn.syntact.data.dao.base;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Update;

import java.util.Collection;

public interface BaseDao<T> {

    @Insert
    Long insert(T t);

    @Insert
    void insert(Collection<T> t);

    @Update
    void update(T t);

    @Delete
    void delete(T t);
}

File: app/src/main/java/com/alexkn/syntact/data/dao/base/BaseDao.kt
package com.alexkn.syntact.data.dao.base

import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Update

interface BaseDao<T> {

    @Insert(onConflict = OnConflictStrategy.ABORT)
    fun insert(t: T): Long

    @Insert(onConflict = OnConflictStrategy.ABORT)
    fun insert(t: Collection<T>)

    @Update
    fun update(t: T)

    @Delete
    fun delete(t: T)
}
